src/Bicep.Core/Registry/Providers/TypesV1Archive.cs

```diff
@@ -44,4 +49,21 @@ private static async Task AddFileToTar(TarWriter tarWriter, string archivePath,
 
         await tarWriter.WriteEntryAsync(tarEntry);
     }
+
+    private static HashSet<string> getAllUniqueTypePaths(string pathToIndex)
+    {
+        FileStream indexStream = new(pathToIndex, FileMode.Open, FileAccess.Read);
+
+        var index = TypeSerializer.DeserializeIndex(indexStream);
+
+        HashSet<string> uniqueTypePaths = new();
+
+        foreach (var typeInformation in index.Resources)
+        {
+            uniqueTypePaths.Add(Path.GetDirectoryName(pathToIndex) + "\\" + typeInformation.Value.RelativePath);
+        }
+
+        return uniqueTypePaths;
```

Comment:
If you take the above suggestion and avoid embedding the file path of the parent directory, you could simplify this whole block by returning an `IEnumerable<string>` instead of `HashSet<string>`:
```suggestion
        return index.Resources.Values.Select(x => x.RelativePath).Distinct();
```